### PR TITLE
Rename AMCL-related files, directories, and namespaces to match proje…

### PR DESCRIPTION
**Pull request by mjeronimo**
Rename AMCL-related files, directories, and namespaces to match project convention.

### CODE REVIEW

File: nav2_amcl_localizer/package.xml [hunk 1]
```diff
@@ -15,14 +15,14 @@
...
   </description>
-  <url>http://wiki.ros.org/amcl</url>
-  <author>Brian P. Gerkey</author>
-  <author>contradict@gmail.com</author>
   <maintainer email="davidvlu@gmail.com">David V. Lu!!</maintainer>
   <maintainer email="mfergs7@gmail.com">Michael Ferguson</maintainer>
   <maintainer email="ahoy@fetchrobotics.com">Aaron Hoy</maintainer>
   <maintainer email="mohammad.haghighipanah@intel.com">Mohammad Haghighipanah</maintainer>
   <license>LGPL</license>
+  <url>http://wiki.ros.org/amcl</url>
```
Comment (mjeronimo) [inline nav2_amcl_localizer/package.xml]: Moved these lines to per xmllint

File: nav2_amcl_localizer/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.5)
-project(nav2_amcl)
+project(nav2_amcl_localizer)
```
Comment (mhpanah) [inline nav2_amcl_localizer/CMakeLists.txt]: Suffix `_localizer` is redundant.

Comment (mikeferguson) [reply nav2_amcl_localizer/CMakeLists.txt]: Also a very long name -- nav2_amcl seems sufficient for a name

**Comment by mhpanah** [comment]
If we really want to change the name for project convention how about using `nav2_amc_localizer`?

**Comment by mikeferguson** [comment]
@mhpanah that would lack the searchability for the term "AMCL" which is what most people know as a name. Why would we want to change away from AMCL?

**Comment by mjeronimo** [comment]
@SteveMacenski @mikeferguson @mhpanah @crdelsey  I was trying to make this module follow some project conventions. That is, for modules we've got a pattern of nav2_<module_name>_<module_type>. For example, nav2_dijkstra_planner or nav2_dwb_controller. The case of amcl presents a bit of a problem because the term localizer/localization appears in both the name (AMCL) and type (localizer). The best I could come up with is nav2_amcl_localizer, realizing that there was some redundancy. This isn't that important of a change, obviously, just trying to continue to remove little inconsistencies. This pattern doesn't hold for all modules though. For example, nav2_map_server, so I guess I can just leave the name alone. What do you think?

As far as the namespaces question, I was trying to introduce namespaces to be consistent with other modules.

**Comment by SteveMacenski** [comment]
I agree with @mikeferguson to leave as just `nav2_amcl`. 

From a quick subsampling of 5 of the packages in this metapackage, they all use namespaces to wrap. Can you explain the strategy for removing them in amcl?

**Comment by mjeronimo** [comment]
@SteveMacenski The PR *adds* namespaces, not removes them. I'll close this one and do another that just adds the namespaces without the module change.

**Pull request closed without merging**